### PR TITLE
at-spi2-core: new formula (Linux-only)

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
Needs https://github.com/Homebrew/homebrew-core/pull/76058 first.

File: Formula/at-spi2-core.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    mkdir "build" do
+      system "meson", "--prefix=#{prefix}", "--libdir=#{lib}", ".."
+      system "ninja"
+      system "ninja", "install"
+    end
+  end
+
+  test do
+    system "#{libexec}/at-spi2-registryd", "-h"
```
Comment (MikeMcQuaid) [inline Formula/at-spi2-core.rb:34]: No better test possible? I'm a bit disappointed that most of the new Linux-only formulae being added have minimal testing.

Comment (danielnachun) [reply Formula/at-spi2-core.rb:34]: Yeah, I'm not too happy about the tests either.  I will try to make a better one here.

Comment (danielnachun) [reply Formula/at-spi2-core.rb:34]: I have a much better test for this now which compiles a small C program that includes one of the headers and links to the shared library, but I'm now stuck at the last step.  I want to use `assert_match` to look for `AT-SPI` in this output from running `#{testpath}/test`:
```
(process:18859): dbind-ERROR **: 09:02:38.116: AT-SPI: Couldn't connect to accessibility bus. Is at-spi-bus-launcher running?
```
but `assert_match "AT-SPI", shell_output("#{testpath}/test")` fails with 
```
An exception occurred within a child process:
  Minitest::Assertion: Expected: 0
  Actual: nil
```
I was following other examples I've seen of test like this in homebrew-core, but there must be some simple mistake I'm making.  I can just push the change if that would make it easier to troubleshoot.

Comment (SMillerDev) [reply Formula/at-spi2-core.rb:34]: It probably writes an error to stderr instead of stdout which `shell_output` captures. `&>` is your friend

Comment (danielnachun) [reply Formula/at-spi2-core.rb]: That was it!  Thank you so much.  I've pushed the new and improved test which I think should be in line with what we usually expect in homebrew-core.

File: Formula/at-spi2-core.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          g_print ("(Index, application, application_child_count)=(%d,%s,%d)\\n",
+                   i, atspi_accessible_get_name (app, NULL), atspi_accessible_get_child_count (app, NULL));
+          g_object_unref (app);
+        }
+
+        return 1;
+      }
+    EOS
+
+    pkg_config_cflags = `pkg-config --cflags --libs atspi-2`.chomp.split
```
Comment (SMillerDev) [inline Formula/at-spi2-core.rb:66]:
```suggestion
    pkg_config_cflags = shell_output("pkg-config --cflags --libs atspi-2").chomp.split
```

File: Formula/at-spi2-core.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                   i, atspi_accessible_get_name (app, NULL), atspi_accessible_get_child_count (app, NULL));
+          g_object_unref (app);
+        }
+
+        return 1;
+      }
+    EOS
+
+    pkg_config_cflags = shell_output("pkg-config --cflags --libs atspi-2").chomp.split
+    system ENV.cc, "test.c", *pkg_config_cflags, "-lgobject-2.0", "-o", "test"
```
Comment (MikeMcQuaid) [inline Formula/at-spi2-core.rb:67]:
```suggestion
    safe_system ENV.cc, "test.c", *pkg_config_cflags, "-lgobject-2.0", "-o", "test"
```
or
```suggestion
    assert_equal true, system ENV.cc, "test.c", *pkg_config_cflags, "-lgobject-2.0", "-o", "test"
```

Comment (danielnachun) [reply Formula/at-spi2-core.rb:67]: Should we be using `safe_system` in general for system calls moving forward?  We will have to fix many dozens of tests for Linux and could use that as an opportunity to make this change as well.

Comment (Bo98) [reply Formula/at-spi2-core.rb:67]: No, `system` in formulae is a totally different thing than `system` used elsewhere so I don't think this needs to change.

`Formula#system` is more less equivalent to `Homebrew.safe_system` + logs.

Comment (MikeMcQuaid) [reply Formula/at-spi2-core.rb:67]: @bo98 is right, ignore me!

**Comment by danielnachun** [comment]
I think this is ready to merge now, as the CI is all green!

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**